docs/_getting-started/dev-mode.md

```diff
@@ -1,64 +1,353 @@
-
-Starting geth in dev mode does the following:
+It is often convenient for developers to work in an environment where changes to client or application software can be deployed and tested rapidly and without putting real-world users or assets at risk. For this purpose, Geth has a `--dev` flag that spins up Geth in "developer mode". This creates a single-node Ethereum test network with no connections to any external peers. It exists solely on the local machine. Starting Geth in developer mode does the following:
 
 -   Initializes the data directory with a testing genesis block
--   Sets max peers to 0
--   Turns off discovery by other nodes
--   Sets the gas price to 0
--   Uses the Clique PoA consensus engine with which allows blocks to be mined as-needed without excessive CPU and memory consumption
+-   Sets max peers to 0 (meaning Geth does not search for peers)
+-   Turns off discovery by other nodes (meaning the node is invisible to other nodes)
+-   Sets the gas price to 0 (no cost to send transactions)
+-   Uses the Clique proof-of-authority consensus engine with which allows blocks to be mined as-needed without excessive CPU and memory consumption
 -   Uses on-demand block generation, producing blocks when transactions are waiting to be mined
 
+This configuration enables developers to experiment with Geth's source code or develop new applications without having to sync to a pre-existing public network. Blocks are only mined when there are pending transactions. Developers can break things on this network without affecting other users. This page will demonstrate how to spin up a local Geth testnet and a simple smart contract will be deployed to it using the Remix online integrated development environment (IDE).
+
+## Prerequisites
+
+It is assumed that the user has a working Geth installation (see [installation guide](../_install-and-build/)).
+It would also be helpful to have basic knowledge of Geth and the Geth console. See [Getting Started](../_getting-started/).
+Some basic knowledge of [Solidity](https://docs.soliditylang.org/) and [smart contract deployment](https://ethereum.org/en/developers/tutorials/deploying-your-first-smart-contract/) would be useful.
+
 ## Start Geth in Dev Mode
 
-You can specify a data directory to maintain state between runs using the `--datadir` option, otherwise, databases are ephemeral and in-memory:
+Starting Geth in developer mode is as simple as providing the `--dev` flag. It is also possible to create a realistic block creation frequency by setting `--dev.period 13` instead of creating blocks only when transactions are pending. There are also additional configuration options required to follow this tutorial. 
+
+First, `http` must be enabled so that the Javascript console can be attached to the Geth node, and some namespaces must be specified so that certain functions can be executed from the Javascript console, specifically `eth`, `web3` and `personal`. 
```

Comment:
Ah interesting. You had to use `geth attach http://` because in dev mode there's no datadir to keep the IPC file?

Alternatively you can just do `geth --dev ... console`. Then you won't need to use http for the console.

But just FYI, I'm not asking you to change the docs. Only if you thinks that's clearer.

Reply:
I just added a short sentence to say `console` is a viable alternative.